crates/uv/tests/pip_compile.rs

```diff
@@ -3200,38 +3200,20 @@ fn override_dependency_from_workspace_invalid_syntax() -> Result<()> {
             .arg("pyproject.toml")
             .current_dir(&context.temp_dir)
             , @r###"
-    success: true
-    exit_code: 0
+    success: false
+    exit_code: 2
     ----- stdout -----
-    # This file was autogenerated by uv via the following command:
-    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2024-03-25T00:00:00Z pyproject.toml
-    blinker==1.7.0
-        # via flask
-    click==8.1.7
-        # via flask
-    flask==3.0.0
-        # via example (pyproject.toml)
-    itsdangerous==2.1.2
-        # via flask
-    jinja2==3.1.3
-        # via flask
-    markupsafe==2.1.5
-        # via
-        #   jinja2
-        #   werkzeug
-    werkzeug==3.0.1
-        # via flask
 
     ----- stderr -----
-    warning: Failed to parse `pyproject.toml`: TOML parse error at line 9, column 29
+    error: Failed to parse: `pyproject.toml`
+      Caused by: TOML parse error at line 9, column 29
```

Comment:
Is this behavior change intentional?

Reply:
Yes, if you have an error in your `override-dependencies` we should fail. The previous behavior was wrong IMO.

Reply:
Makes sense to me